### PR TITLE
Add admissions from supervision to prison by violation charts

### LINKED ISSUE #1455: [US_TN] Create the "Admissions from supervision by prison by most severe violation" chart
**Is your feature request related to a problem? Please describe.**
Create the "Admissions from supervision by prison by most severe violation" chart for US_TN on the "Supervision to Prison" page.

metric should be called supervisionToPrisonPopulationByMostSevereViolation, and will be an instance of PopulationSnapshotMetric.
id: "supervisionToPrisonPopulationByMostSevereViolation",
tenantId: this.rootStore.currentTenantId,
compartment: "SUPERVISION",
sourceFilename: "supervision_to_prison_population_snapshot_by_dimension",
rootStore: this.rootStore,
dataTransformer: createDistrictPopulationSnapshot,
accessor: mostSevereViolation,
enabledFilters: [FILTER_TYPES.GENDER, FILTER_TYPES.DISTRICT, FILTER_TYPES.MOST_SEVERE_VIOLATION],
enabledMoreFilters: [FILTER_TYPES.AGE_GROUP, FILTER_TYPES.LEGAL_STATUS]

[Design](https://www.figma.com/file/Mw6kSEP2jdr9i7UVEjPuCI/UP-Consolidation?node-id=862%3A343)

Note: the design shows this as a rate chart (Y-axis is a percentage) but please make it a count chart (Y-axis is a count)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context about the feature request here.

### LINKED ISSUE #1456: [US_TN] Create the "Admissions from supervision by prison by number of violations" chart
**Is your feature request related to a problem? Please describe.**
Create the "Admissions from supervision by prison by number of violations" chart for US_TN on the "Supervision to Prison" page.

metric should be called supervisionToPrisonPopulationByNumberOfViolations, and will be an instance of PopulationSnapshotMetric.
id: "supervisionToPrisonPopulationByNumberOfViolations",
tenantId: this.rootStore.currentTenantId,
compartment: "SUPERVISION",
sourceFilename: "supervision_to_prison_population_snapshot_by_dimension",
rootStore: this.rootStore,
dataTransformer: createDistrictPopulationSnapshot,
accessor: numberOfViolations,
enabledFilters: [FILTER_TYPES.GENDER, FILTER_TYPES.DISTRICT, FILTER_TYPES.MOST_SEVERE_VIOLATION],
enabledMoreFilters: [FILTER_TYPES.AGE_GROUP, FILTER_TYPES.LEGAL_STATUS, FILTER_TYPES.NUMBER_OF_VIOLATIONS]

[Design](https://www.figma.com/file/Mw6kSEP2jdr9i7UVEjPuCI/UP-Consolidation?node-id=3520%3A12398)

Note: the design shows this as a rate chart (Y-axis is a percentage) but please make it a count chart (Y-axis is a count)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

1. Added the "Admissions from supervision by prison by most severe violation" chart
2. Added the "Admissions from supervision by prison by number of violations" chart
3. Fixed filters sorting bug to always put `All` option first

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #1455 
Closes #1456 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by nasaownsky** [comment]
@jovergaag Two things worth mentioning:
1. I haven't find in the designs light color for highlighting filtered bars in golden charts.
2. Also I don't know, how to separate golden charts from greenish. In the designs I noticed a trend where the charts turns gold if there are only few bars, so I've made it that way.

File: src/core/models/PathwaysMetric.ts [hunk 1]
```diff
@@ -114,6 +114,10 @@ export default abstract class PathwaysMetric<RecordFormat extends MetricRecord>
     return this.content.note;
   }
 
+  get legend(): string | undefined {
```
Comment (jovergaag) [inline src/core/models/PathwaysMetric.ts:117]: This isn't exactly a legend, and we may eventually have a legend on these charts so we should probably reserve that name. `chartXAxisTitle` is probably what we want here.

File: src/core/VizPopulationSnapshot/VizPopulationSnapshot.tsx [hunk 2]
```diff
@@ -125,11 +142,14 @@ const VizPopulationSnapshot: React.FC<VizPopulationOverTimeProps> = ({
           axes={[
             {
               orient: "left",
-              ticks: 3,
+              ticks: 4,
```
Comment (jovergaag) [inline src/core/VizPopulationSnapshot/VizPopulationSnapshot.tsx:145]: I just noticed that sometimes the bars extend above the highest tick which makes it hard to read sometimes. This exists for other charts as well in addition to these new charts. Can you please find a way to make sure that the highest tick is above the highest bar. 

Screen Shot 2021-12-02 at 11 06 20 AM

Screen Shot 2021-12-02 at 11 08 10 AM

**Review by jovergaag — CHANGES_REQUESTED** [review]
Wow, I love how easily these two charts came together. Fantastic work! 

I know there has been a lot of churn the last few days as I have been renaming metrics and reorganizing things but I think its starting to pay off. Thank you for your patience :)

File: src/core/CoreStore/MetricsStore.ts [hunk 3]
```diff
@@ -77,6 +77,10 @@ export default class MetricsStore {
         [PATHWAYS_SECTIONS.countOverTime]: this.supervisionToPrisonOverTime,
         [PATHWAYS_SECTIONS.countByLocation]: this
           .supervisionToPrisonPopulationByDistrict,
+        [PATHWAYS_SECTIONS.countByViolation]: this
+          .supervisionToPrisonPopulationByMostSevereViolation,
+        [PATHWAYS_SECTIONS.countByNumber]: this
```
[the remark this answers is not in the document]
Comment (jovergaag) [reply src/core/CoreStore/MetricsStore.ts:82]: Once you change the name of these I'll sync the content.

**Comment by nasaownsky** [comment]
@jovergaag Most of the highest ticks now is above the highest bar, but not always as you can see on `Admissions from supervision by district` chart. This is Semiotic-spesific, and I have not found a better solution.

File: src/utils/formatStrings.ts [hunk 4]
```diff
@@ -228,6 +229,11 @@ function decrypt(hexString: string): string {
   return plainText;
 }
 
+const roundToDecimals = (value: number): number => {
```
Comment (jovergaag) [inline src/utils/formatStrings.ts:232]: I think `getCeiling` is more accurate here. Great logic!

**Review by jovergaag — CHANGES_REQUESTED** [review]

File: src/core/VizCountOverTimeWithAvg/VizCountOverTimeWithAvg.tsx [hunk 5]
```diff
@@ -148,7 +151,7 @@ const VizCountOverTimeWithAvg: React.FC<VizCountOverTimeWithAvgProps> = ({
         axes={[
           {
             orient: "left",
-            ticks: 3,
+            ticks: 5,
```
[the remark this answers is not in the document]
Comment (nasaownsky) [reply src/core/VizCountOverTimeWithAvg/VizCountOverTimeWithAvg.tsx:153]: Great! I made a couple of fixes with this in mind.

File: src/utils/formatStrings.ts [hunk 6]
```diff
@@ -229,9 +229,19 @@ function decrypt(hexString: string): string {
   return plainText;
 }
 
-const roundToDecimals = (value: number): number => {
+const getTicks = (value: number) => {
```
Comment (jovergaag) [inline src/utils/formatStrings.ts]: YES, the ticks are so much better now! Thank you.

**Review by jovergaag — APPROVED** [review]

**Pull request merged**